src/algokit/cli/tasks/sign_transaction.py

```diff
@@ -0,0 +1,63 @@
+import json
+from pathlib import Path
+from typing import cast
+
+import click
+from algosdk import encoding
+from algosdk.transaction import Transaction, retrieve_from_file, write_to_file
+
+from algokit.cli.tasks.utils import get_account_with_private_key
+
+
+def get_transaction(file: Path | None, transaction: str | None) -> Transaction:
+    if file:
+        txns: list[Transaction] = retrieve_from_file(str(file))  # type: ignore[no-untyped-call]
+        if len(txns) > 1:
+            raise click.ClickException("Only one transaction per file is supported.")
+        return txns[0]
+    elif transaction:
+        return cast(Transaction, encoding.msgpack_decode(transaction))  # type: ignore[no-untyped-call]
```

Comment:
Does this take base 64?

Reply:
I more meant `transaction` itself should be a base64 string, is that the right input for that `msgpack_decode` method?

Reply:
Also, if we use cast do we need to do some validation to check it is intact? Does it matter?

Reply:
all unsigned transaction objects should be inheriting from Transaction parent class. In case its already signed there is a separate validator that will throw an exception if user provided something that is signed already.